globus_sdk/auth/client_types/base.py

```diff
@@ -187,7 +187,7 @@ def oauth2_exchange_code_for_tokens(self, auth_code):
         return self.current_oauth2_flow_manager.exchange_code_for_tokens(
             auth_code)
 
-    def oauth2_refresh_token(self, refresh_token):
+    def oauth2_refresh_token(self, refresh_token, additional_params=None):
```

Comment:
I will never object to extension points like this, as they're exposing underlying API functionality that already exists.
So, this is good and we should keep it.

My one question is: why do you need it? I know you said that you needed to send the `client_id` for the Native App, but the `NativeAppAuthClient` class provides that customization already. Is there a reason you need that capability here as well?